### ISSUE TITLE
Building net on GPU fails with error "problem in cublas create"

### ISSUE THREAD

**Issue by Andrea-Oliveri** [deephealthproject/eddl#136]
Good Morning,
I was trying to run my program on GPU. I successfully compiled and installed eddl-GPU and I compiled my program. However, when I try to run it, execution fails on the function
```
build(net, sgd(0.01, 0.9, 0.0, true), {"cross_entropy"}, {"categorical_accuracy"}, CS_GPU({1}, "full_mem"));
```
with error:

> problem in cublas create

Knowing that when using `CS_CPU(4, "full_mem")` the program runs without failing, how could I fix this error?
Thank you in advance for your help.

**Comment by RParedesPalacios** [comment]
This error probably comes from not enough memory.

**Comment by Andrea-Oliveri** [comment]
Unfortunately I don't think that is the case: the GPU I am using has 15079MiB of internal memory, and execution fails even when I use `CS_GPU({1}, "low_mem")`

**Comment by RParedesPalacios** [comment]
Ok, please copy paste here *all* the code

**Comment by Andrea-Oliveri** [comment]
Even just the exemple 7_mnist_conv.cpp does not work on my machine with cuda-10.2: when run, it returns this:
```
Generating Random Table
CS with low memory setup
Selecting GPU device 0
EDDLL is running on GPU device 0, ^A
.
.
.
.
.
.
.
.
.
.
terminate called after throwing an instance of 'std::runtime_error'
  what():  problem in cublas create
Aborted (core dumped)
```

**Comment by RParedesPalacios** [comment]
Andrea, this is a strange behaviour that sometimes appeared in the past. It is a problem trying to initialise cublas. Each "." is an attempt to initialise that fails. It is something that doesn't use to appear now and we thought it was solved. Right now my guess is that you can try other cuda version. I will try to come back to this issue later.

**Comment by Andrea-Oliveri** [comment]
I tried changing cuda version and indeed does work for me with cuda-10.0 and gcc7, but not on cuda-10.2 and gcc8. Thank you for your help

**Comment by RParedesPalacios** [comment]
BTW, check that this is also happening in **TensorFlow and PyTorch**, and the solution that some users propose is to downgrade to 10.1:

https://github.com/tensorflow/tensorflow/issues/9489#issuecomment-557606158

**Comment by diegobenedicto** [comment]
I had the same error and I have solved it downgrading libcublas10 from 10.2.2.89-1 to 10.2.1.243-1, as indicated in https://github.com/tensorflow/tensorflow/issues/9489#issuecomment-616949644
I use cuda-10.1 and gcc8
